cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -670,30 +669,30 @@ fromShelleyTxOut era ledgerTxOut =
 
     ShelleyBasedEraBabbage -> error "TODO: Babbage"
 
-toAlonzoTxOutDataHash :: TxOutDatum CtxUTxO era
-                      -> StrictMaybe (Alonzo.DataHash StandardCrypto)
+-- TODO: If ledger creates an open type family for datums
+-- we can consolidate this function with the Babbage version
+toAlonzoTxOutDataHash
+  :: TxOutDatum CtxUTxO AlonzoEra
+  -> StrictMaybe (Alonzo.DataHash StandardCrypto)
 toAlonzoTxOutDataHash  TxOutDatumNone                        = SNothing
 toAlonzoTxOutDataHash (TxOutDatumHash _ (ScriptDataHash dh)) = SJust dh
-toAlonzoTxOutDataHash (TxOutInlineDatum _ sd) =
-  let ScriptDataHash alonzoDataHash = hashScriptData sd
-  in SJust alonzoDataHash
+toAlonzoTxOutDataHash (TxOutDatumInline inlineDatumSupp _sd) =
+  case inlineDatumSupp :: InlineDatumSupportedInEra AlonzoEra of {}
```

Comment:
Can you explain this change?

Reply:
Sure so basically in the AlonzoEra we can only have datum hashes or nothing. Whereas in Babbage era we can have inline datums, datum hashes or nothing. Because there is no open type family for datums, I need to define two separate functions that return the correct type based on the era. I've asked the ledger team to define an open type family for datums and they will probably do this after the hardfork. When they do this, we will be able to have a single function that instantiates the correct type based on the `ShelleyBasedEra era` we feed it.

Reply:
`case inlineDatumSupp :: InlineDatumSupportedInEra AlonzoEra of {}` tells the compiler it's not possible to pattern match `TxOutDatumInline` because only `InlineDatumSupportedInEra BabbageEra` exists.

Reply:
Oh I see.  Thanks!